docs/developers/library-structure.md

```diff
@@ -0,0 +1,39 @@
+# Library Structure
+
+The kcp library is set up and initialized in several phases. This document describes those phases and the responsibilities of each.
+
+```go
+opts := options.NewOptions()
+completedOptions := opts.Complete()
+conf := config.NewConfig(completedOptions)
+completedConfig := conf.Complete()
+srv := server.NewServer(completedConfig)
+srv.Init()
+srv.Run(ctx)
+```
+
+## Options
+
+Options parses basic configuration options, such as flags or environment variables. `NewOptions()` does default parsing
+of environment variables and CLI flags and sets defaults. You then can change options properties
```

Comment:
this is not entirely correct. `NewOptions` only returns default value. `options.AddFlags` and then `pflag.ParseFlags` will do the command line parsing.